dist/src/main/java/io/camunda/application/commons/identity/WebSessionRepositoryConfiguration.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.application.commons.identity;
+
+import io.camunda.authentication.session.ConditionalOnPersistentWebSessionEnabled;
+import io.camunda.authentication.session.WebSessionDeletionTask;
+import io.camunda.authentication.session.WebSessionMapper;
+import io.camunda.authentication.session.WebSessionMapper.SpringBasedWebSessionAttributeConverter;
+import io.camunda.authentication.session.WebSessionRepository;
+import io.camunda.exporter.config.ConnectionTypes;
+import io.camunda.search.clients.DocumentBasedSearchClient;
+import io.camunda.search.clients.DocumentBasedWriteClient;
+import io.camunda.search.clients.PersistentSessionSearchClient;
+import io.camunda.search.clients.PersistentWebSessionSearchClientImpl;
+import io.camunda.search.connect.configuration.ConnectConfiguration;
+import io.camunda.webapps.schema.descriptors.usermanagement.index.PersistentWebSessionIndexDescriptor;
+import io.camunda.zeebe.util.error.FatalErrorHandler;
+import jakarta.servlet.http.HttpServletRequest;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.convert.support.GenericConversionService;
+import org.springframework.session.config.annotation.web.http.EnableSpringHttpSession;
+
+@Configuration
+@EnableSpringHttpSession
+@ConditionalOnPersistentWebSessionEnabled
```

Comment:
All Spring Beans (except the REST Gateway, which is based on the Spring MVC, etc.) declarations belong in the `dist` module. The `WebSecurityConfig` would need to be moved there as well. I don't know why this has not happened because I didn't implement it. You could tackle this in a separate PR; I will not touch that with this PR.